src/Jenssegers/Mongodb/Connection.php

```diff
@@ -136,6 +136,14 @@ protected function createConnection($dsn, array $config, array $options)
             $driverOptions = $config['driver_options'];
         }
 
+        // Check if the credentials are not already set in the options
+        if (!isset($options['username'])) {
```

Comment:
Yes, I've seen that that is indeed where the problem was.

I've got the test suite running locally now as well so I can test before pushing bad code again.
I want to test the entire suite before pushing so I just need to get MySQL up and running, after that I will submit the new changes again